pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +192,202 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def get_transformation_block(
+    in_channels: int,
+    kernel_size: Union[Tuple[int, int], int],
+    stride: Union[Tuple[int, int], int],
+    padding: Union[Tuple[int, int], int],
+    impl_params: bool = True,
+) -> nn.Module:
+    if impl_params:
+        return nn.AvgPool2d(kernel_size=kernel_size, stride=stride, padding=padding)
+    return nn.Conv2d(in_channels, 3, kernel_size, stride=stride, padding=padding)
```

Comment:
Are you sure about this? `Conv2d` and `AvgPool2d` are quite different blocks. If this is the case please add the link to the reference implementation.

Reply:
The authors also describe this (See [here](https://github.com/pmeier/adaptive-style-transfer/blob/07a3b3fcb2eeed2bf9a22a9de59c0aea7de44181/module.py#L238)).

> This is a simplified version of transformer block described in our paper

Reply:
So instead of the conv block they simply use an avg pool? :smile:

Reply:
Right. I suppose that the convolution has led to problems, because the parameters can be learned additionally during the training.